src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -122,11 +125,14 @@ private void processDirectors() {
                 if (!(tnt.getVelocity().lengthSquared() > 0.35) || tracking.containsKey(tnt)) {
                     continue;
                 }
-                Craft c = CraftManager.getInstance().fastNearestCraftToLoc(tnt.getLocation());
-                tracking.put(tnt, tnt.getVelocity().lengthSquared());
-                if (c == null) {
+
+                Craft c = getDirectingCraft(tnt);
+                if(c == null) {
                     continue;
                 }
+
+                tracking.put(tnt, tnt.getVelocity().lengthSquared());
```

Comment:
In the case where the directing craft is null, the server will constantly run getDirectingCraft (an expensive function call) on the TNT.  Leaving this before the null will reduce server lag by a significant margin.